src/main/java/com/google/sps/listeners/ConnectionPoolContextListener.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.sps.listeners;
+
+import com.google.sps.classes.Utility;
+import com.zaxxer.hikari.HikariConfig;
+import com.zaxxer.hikari.HikariDataSource;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import javax.servlet.ServletContext;
+import javax.servlet.ServletContextEvent;
+import javax.servlet.ServletContextListener;
+import javax.servlet.annotation.WebListener;
+import javax.sql.DataSource;
+
+@WebListener("Creates a connection pool that is stored in the Servlet's context for later use.")
+public class ConnectionPoolContextListener implements ServletContextListener {
+
+  // Saving credentials in environment variables is convenient, but not secure - consider a more
+  // secure solution such as https://cloud.google.com/kms/ to help keep secrets safe.
+  private static final String 
+      CLOUD_SQL_CONNECTION_NAME = "internship-platform-step-2020:us-central1:mintern-instance";
+  private static final String DB_USER = "root";
+  private static final String DB_PASS = "mintern";
+  private static final String DB_NAME = "Mintern";
+
+  private DataSource createConnectionPool() {
+    // [START cloud_sql_mysql_servlet_create]
+    // The configuration object specifies behaviors for the connection pool.
+    HikariConfig config = new HikariConfig();
+
+    // Configure which instance and what database user to connect with.
+    config.setJdbcUrl(String.format("jdbc:mysql:///%s", DB_NAME));
+    config.setUsername(DB_USER); // e.g. "root", "postgres"
+    config.setPassword(DB_PASS); // e.g. "my-password"
+
+    // maximumPoolSize limits the total number of concurrent connections this pool will keep. Ideal
+    // values for this setting are highly variable on app design, infrastructure, and database.
+    config.setMaximumPoolSize(20);
```

Comment:
Consider setting these values as constants to more easily change them in the future. Same with other integer values.

Reply:
Done!